invenio_vocabularies/datastreams/readers.py

```diff
@@ -224,19 +224,31 @@ def _iter(self, fp, *args, **kwargs):
 class XMLReader(BaseReader):
     """XML reader."""
 
+    def __init__(self, root_element=None, *args, **kwargs):
+        """Constructor."""
+        # TODO: How to make root_element mandatory?
+        self.root_element = root_element
+        super().__init__(*args, **kwargs)
+
     def _iter(self, fp, *args, **kwargs):
         """Read and parse an XML file to dict."""
         # NOTE: We parse HTML, to skip XML validation and strip XML namespaces
         record = None
         try:
             xml_tree = fromstring(fp)
-            record = etree_to_dict(xml_tree).get("record")
+            xml_dict = etree_to_dict(xml_tree)
         except Exception as e:
             xml_tree = html_parse(fp).getroot()
-            record = etree_to_dict(xml_tree)["html"]["body"].get("record")
+            xml_dict = etree_to_dict(xml_tree)["html"]["body"]
 
-        if not record:
-            raise ReaderError(f"Record not found in XML entry.")
+        if self.root_element:
```

Comment:
Looks duplicated from transformer. Why not doing this in only 1 place?

Reply:
There is:
* One XML reader, used in several places
* One XML transformer, not used anywhere

We could (by order of preference):
1. Leave it as is
2. Delete the XML transformer
3. Put this logic in a common class

Reply:
Let's leave if for another day. It feels like the transformer should do something else.